pkg/codegen/schema/bind.go

```diff
@@ -485,15 +485,19 @@ func (spec *PackageSpec) validateTypeToken(allowedPackageNames map[string]bool,
 	diags := hcl.Diagnostics{}
 
 	path := memberPath(section, token)
-	var packageName string
-	if i := strings.Index(token, ":"); i != -1 {
-		packageName = token[:i]
+	parts := strings.Split(token, ":")
+	if len(parts) > 3 {
+		err := errorf(path, "invalid token '%s' (should have at most three parts)", token)
+		diags = diags.Append(err)
 	}
-	if !allowedPackageNames[packageName] {
-		error := errorf(path, "invalid token '%s' (must have package name '%s')", token, spec.Name)
-		diags = diags.Append(error)
+	if !allowedPackageNames[parts[0]] {
+		err := errorf(path, "invalid token '%s' (must have package name '%s')", token, spec.Name)
+		diags = diags.Append(err)
+	}
+	if (parts[1] == "" || strings.EqualFold(parts[1], "index")) && strings.EqualFold(parts[2], "provider") {
```

Comment:
I think `parts[2]` will panic if the input is "name:"

Reply:
good point, that's also an invalid token but we should return a nice error rather than panic. I'll add a check and test for that.